### PR TITLE
feat(api): Ensure org_id checking including cached responses

### PR DESCRIPTION
**Pull request by chambridge**
# Overview

* Use Flask's after_request feature to perform the check_org_id check on defined endpoints
* Remove decorator that didn't execute when caching was performed

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -56,6 +57,30 @@
 
 custom_filter_fields = ["operating_system"]
 
+ORG_ID_CHECK_ENDPOINTS = [
```
Comment (thearifismail) [inline app/__init__.py:60]: Isn't there some swagger api to provide alll endpoints? 
Where are these endpoint strings coming from?

**Review by fstavela — APPROVED** [review]

**Review by chambridge — COMMENTED** [review]
@thearifismail I tried to provide further context

File: app/__init__.py (hunk 1 above)
Comment (thearifismail) [reply app/__init__.py:60]: I see, even `api_` is not needed.  The potential problem with manually building the endpoint list is that someone will forget to add some new endpoint leaving the potential for data exposure.

**Review by thearifismail — APPROVED** [review]
